frontend/fingertips-frontend/components/views/ViewsContext.tsx

```diff
@@ -0,0 +1,46 @@
+import OneIndicatorOneAreaView from './OneIndicatorOneAreaView';
+import {
+  SearchParams,
+  SearchStateManager,
+  SearchStateParams,
+} from '@/lib/searchStateManager';
+import OneIndicatorTwoOrMoreAreasView from './OneIndicatorTwoOrMoreAreasView';
+import TwoOrMoreIndicatorsAreasView from './TwoOrMoreIndicatorsAreasView';
+import { areaCodeForEngland } from '@/lib/chartHelpers/constants';
+import TwoOrMoreIndicatorsEnglandView from './TwoOrMoreIndicatorsEnglandView';
+import { JSX } from 'react';
+
+type ViewsContextProps = {
+  searchState: SearchStateParams;
+};
+
+function viewSelector(
+  areaCodes: string[],
+  indicators: string[],
+  searchState: SearchStateParams
+): JSX.Element | undefined {
+  if (indicators.length === 1) {
+    if (areaCodes.length === 1) {
+      return <OneIndicatorOneAreaView searchState={searchState} />;
+    } else if (areaCodes.length >= 2) {
+      return <OneIndicatorTwoOrMoreAreasView searchState={searchState} />;
+    }
+  } else if (indicators.length >= 2) {
+    if (areaCodes.length === 1 && areaCodes[0] === areaCodeForEngland) {
+      return <TwoOrMoreIndicatorsEnglandView searchState={searchState} />;
+    }
+    return <TwoOrMoreIndicatorsAreasView searchState={searchState} />;
+  }
```

Comment:
Just a suggestion, and I'm happy for you to decide whether you want to implement it or not, but I think this might be a bit more readable if you avoid nesting the if statements. Something like:
```suggestion
  if (indicators.length === 1 && areaCodes.length === 1) {
    return <OneIndicatorOneAreaView searchState={searchState} />;
  }

  if (indicators.length === 1 && areaCodes.length >= 2) {
    return <OneIndicatorTwoOrMoreAreasView searchState={searchState} />;
  }

  if (
    indicators.length >= 2 &&
    areaCodes.length === 1 &&
    areaCodes[0] === areaCodeForEngland
  ) {
    return <TwoOrMoreIndicatorsEnglandView searchState={searchState} />;
  }

  return <TwoOrMoreIndicatorsAreasView searchState={searchState} />;
```

I think it would lend itself better to the addition of more views in future as well.

Reply:
I've tried both versions, so happy to go with the more maintainable option.